src/Common/src/Internal/Runtime/EETypeOptionalFieldsBuilder.cs

```diff
@@ -93,7 +93,15 @@ public override string ToString()
 
             for (int i = 0; i < (int)EETypeOptionalFieldsElement.Count; i++)
             {
-                sb.Append(_rgFields[i]._value.ToString());
+                if (_rgFields[i]._fieldPresent)
+                {
+                    sb.Append(_rgFields[i]._value.ToString());
```

Comment:
Mmm. It is a piece of very reusable code. If you don't want to, you don't need to do this in the pull request, but I think eventually we will end up with it being in common. We will need to put it in a namespace that is easily accessible from everywhere though (maybe even `System`).